File: javascript/gemini/p124_nc.js
// 1. Arg count validated (process.argv includes node path and script path)
if (process.argv.length !== 3) {
    console.error("Error: Exactly one argument required.");
    process.exit(1);
}

// 2. Get argument
let inputVal = process.argv[2];

// 3. Print argument
console.log(`Original: '${inputVal}'`);

// 4. Trim whitespace
inputVal = inputVal.trimEnd();

// 5. Print again
console.log(`Trimmed:  '${inputVal}'`);

// 6. Exit
process.exit(0);

File: java/gemini/p124_nc.java
public class p124_nc {
    public static void main(String[] args) {
        // 1. Arg count validated
        if (args.length != 1) {
            System.err.println("Error: Exactly one argument required.");
            System.exit(1);
        }

        // 2. Get argument
        String inputVal = args[0];

        // 3. Print argument
        System.out.println("Original: '" + inputVal + "'");

        // 4. Trim whitespace
        // Note: stripTrailing() available in Java 11+
        inputVal = inputVal.stripTrailing();

        // 5. Print again
        System.out.println("Trimmed:  '" + inputVal + "'");

        // 6. Exit
        System.exit(0);
    }
}